### PR TITLE
Add RPC authentication

### LINKED ISSUE #11: Implement rpc-auth
Needs https://github.com/steemit/koa-jsonrpc/pull/1

### PR DESCRIPTION
**Pull request by jnordberg**
Closes #11

### CODE REVIEW

File: src/drafts.ts [hunk 1]
```diff
@@ -6,53 +6,56 @@
...
-async function readDrafts(username: string): Promise<any[]> {
-    return await store.readJSON(draftsKey(username)) || []
+async function readDrafts(account: string): Promise<any[]> {
+    return await store.readJSON(draftsKey(account)) || []
 }
 
-async function writeDrafts(username: string, drafts: any) {
-    await store.writeJSON(draftsKey(username), drafts)
+async function writeDrafts(account: string, drafts: any) {
+    await store.writeJSON(draftsKey(account), drafts)
```
Comment (goldibex) [inline src/drafts.ts]: `return await`

File: test/common.ts [hunk 2]
```diff
@@ -20,16 +21,36 @@ export class RPCError extends Error {
...
+    async signedCall(method: string, signer: RPCSigner, ...params) {
+        const request = signRequest(this.buildRequest(method, ...params), signer.account, [signer.key])
+        const response = await this.send(request)
+        return this.resolveResponse(response)
+    }
+    private async send(data: any) {
+        return await utils.jsonRequest(this.requestOpts, data)
+    }
+    private buildRequest(method: string, ...params): any {
+        return {id: ++this.seqNo, jsonrpc: '2.0', method, params}
```
Comment (goldibex) [inline test/common.ts]: don't use preincrements

File: test/feature-flags.ts [hunk 3]
```diff
@@ -67,7 +81,7 @@ describe('feature flags', function() {
             result[flag] = Math.round((distr[flag] / total) * 100)
         }
         // close enough
```
Comment (goldibex) [inline test/feature-flags.ts]: 😰

File: src/drafts.ts (hunk 1 above)
Comment (jnordberg) [reply src/drafts.ts]: the missing return is not accidental here, we don't want to return the store metadata to users

File: test/feature-flags.ts (hunk 3 above)
Comment (jnordberg) [reply test/feature-flags.ts]: it's spot on now when i'm testing 50% of the names instead of sampling every 50th ;)

**Pull request merged**